efficiency/python/lst_plot_performance.py

```diff
@@ -0,0 +1,816 @@
+        types = [args.objecttype]
+
+    if args.compare:
+        types = objecttype_choices
+
+    if args.selection:
+        sels["eff"] = [args.selection]
+
+    if args.pdgid != None:
+        pdgids["eff"] = [args.pdgid]
+        pdgids["fakerate"] = [args.pdgid]
+        pdgids["duplrate"] = [args.pdgid]
+
+    if args.charge != None:
+        charges["eff"] = [args.charge]
+        charges["fakerate"] = [args.charge]
+        charges["duplrate"] = [args.charge]
+
+    if args.variable:
+        # dxy and dz are only in efficiency
+        if args.variable != "dxy" and args.variable != "dz":
+            variables["eff"] = [args.variable]
+            variables["fakerate"] = [args.variable]
+            variables["suplrate"] = [args.variable]
+        else:
+            variables["eff"] = [args.variable]
+            variables["fakerate"] = []
+            variables["suplrate"] = []
+
+    if args.individual:
+        # Only eff / TC matters here
+        breakdowns["eff"]["TC"] = [False]
+        breakdowns["fakerate"]["TC"] = [False]
+        breakdowns["duplrate"]["TC"] = [False]
+    else:
+        # Only eff / TC matters here
+        breakdowns["eff"]["TC"] = [True]
+        breakdowns["fakerate"]["TC"] = [True]
+        breakdowns["duplrate"]["TC"] = [True]
+
+    if args.yzoom:
+        args.yzooms = [args.yzoom]
+
+    if args.xcoarse:
+        args.xcoarses = [args.xcoarse]
+
+    for metric in metrics:
+        for sel in sels[metric]:
+            for variable in variables[metric]:
+                for yzoom in yzooms:
+                    for xcoarse in xcoarses[variable]:
+                        for typ in types:
+                            for breakdown in breakdowns[metric][typ]:
+                                for pdgid in pdgids[metric]:
+                                    for charge in charges[metric]:
+                                        args.metric = metric
+                                        args.objecttype = typ
+                                        args.selection = sel
+                                        args.pdgid = pdgid
+                                        args.charge = charge
+                                        args.variable = variable
+                                        args.breakdown = breakdown
+                                        args.yzoom = yzoom
+                                        args.xcoarse = xcoarse
+                                        print(args)
+                                        plot(args)
+
+#______________________________________________________________________________________________________
+def compare(file1, file2, legend0, legend1):
+
+    f1 = r.TFile(file1)
+    f2 = r.TFile(file2)
+
+    h1 = f1.Get("githash").GetTitle()
+    h2 = f2.Get("githash").GetTitle()
+
+    i1 = f1.Get("input").GetTitle()
+    i2 = f2.Get("input").GetTitle()
+
+    params = {}
+    params["output_dir"] = "comparison/{}_v_{}".format(h1, h2)
```

Comment:
Could you elaborate a bit more..?

Reply:
Let's imagine that I have three efficiencies with the tags "ref", "test1" and "test2". Suppose that I want to run the comparison "test1" vs. "ref" and "test2" vs. "ref". Do I understand correctly that for both comparisons the plots will end up in a folder named "comparison", i.e. the comparison folder has no info on the tags that I compared? If that's the case, I was thinking that it would be better to include the info on the tags compared in the folder name. That could spare us from mistakenly overwriting comparison folders as well, because, e.g. we forgot to manually rename the file before running another comparison.

Or maybe I didn't read the code well and folder names of comparisons are clearly distinguished by some identifier?

Reply:
Actually this function is not used.
And the actualy compare option will put things in something like

performance/(tag)_githash-sample_githash-sample/

So it should be alright

Reply:
Since we compare, shouldn't there be two tags in the names? And if the function is not used, should we consider removing it to avoid confusion? In any case, I am resolving this comment, thanks for the clarifications.